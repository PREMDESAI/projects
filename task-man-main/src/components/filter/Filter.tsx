import { getStatusText } from '@/helpers';
import { TaskStatus, useTasks } from '@/hooks/use-tasks';
import { Flex, Select, TextField } from '@radix-ui/themes';
import React from 'react';

export const Filter: React.FC = () => {
  const { filterTasks, statuses } = useTasks();
  const [filterText, setFilterText] = React.useState<string>();
  const [filterStatus, setFilterStatus] = React.useState<string>();

  React.useEffect(() => {
    const status = (
      filterStatus
        ? filterStatus === 'all'
          ? undefined
          : filterStatus
        : undefined
    ) as TaskStatus | undefined;

    filterTasks({ text: filterText, taskStatus: status });
  }, [filterText, filterStatus]);

  return (
    <Flex direction='row' gap='1' style={{ padding: '12px 0', width: '100%' }}>
      <TextField.Root
        placeholder='Filter tasks...'
        onChange={(event) => setFilterText(event.target.value)}
        style={{ minWidth: '158px', maxWidth: '158px' }}
      />
      <Select.Root
        defaultValue='all'
        onValueChange={(value) => setFilterStatus(value as TaskStatus)}
      >
        <Select.Trigger style={{ width: '158px' }} />
        <Select.Content>
          <Select.Item value='all'>ALL</Select.Item>
          {statuses.map((status) => (
            <Select.Item value={status} key={`filter-status-${status}`}>
              {getStatusText(status)}
            </Select.Item>
          ))}
        </Select.Content>
      </Select.Root>
    </Flex>
  );
};
