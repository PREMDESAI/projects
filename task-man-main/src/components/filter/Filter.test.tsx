import { render } from '@testing-library/react';
import { Theme } from '@radix-ui/themes';
import { describe, it } from 'vitest';

import { Filter } from './Filter';

describe('Filter', () => {
  it('renders', async () => {
    const { container } = render(
      <Theme>
        <Filter />
      </Theme>
    );
    expect(container).toMatchSnapshot();
  });
});
