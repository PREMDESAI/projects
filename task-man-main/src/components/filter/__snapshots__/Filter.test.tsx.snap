// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter > renders 1`] = `
<div>
  <div
    class="radix-themes"
    data-accent-color="indigo"
    data-gray-color="slate"
    data-has-background="true"
    data-is-root-theme="true"
    data-panel-background="translucent"
    data-radius="medium"
    data-scaling="100%"
  >
    <div
      class="rt-Flex rt-r-fd-row rt-r-gap-1"
      style="padding: 12px 0px; width: 100%;"
    >
      <div
        class="rt-TextFieldRoot rt-r-size-2 rt-variant-surface"
        style="min-width: 158px; max-width: 158px;"
      >
        <input
          class="rt-reset rt-TextFieldInput"
          placeholder="Filter tasks..."
          spellcheck="false"
        />
      </div>
      <button
        aria-autocomplete="none"
        aria-controls="radix-:r0:"
        aria-expanded="false"
        class="rt-reset rt-SelectTrigger rt-r-size-2 rt-variant-surface"
        data-state="closed"
        dir="ltr"
        role="combobox"
        style="width: 158px;"
        type="button"
      >
        <span
          class="rt-SelectTriggerInner"
        >
          <span
            style="pointer-events: none;"
          >
            ALL
          </span>
        </span>
        <svg
          aria-hidden="true"
          class="rt-SelectIcon"
          fill="currentcolor"
          height="9"
          viewBox="0 0 9 9"
          width="9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.135232 3.15803C0.324102 2.95657 0.640521 2.94637 0.841971 3.13523L4.5 6.56464L8.158 3.13523C8.3595 2.94637 8.6759 2.95657 8.8648 3.15803C9.0536 3.35949 9.0434 3.67591 8.842 3.86477L4.84197 7.6148C4.64964 7.7951 4.35036 7.7951 4.15803 7.6148L0.158031 3.86477C-0.0434285 3.67591 -0.0536285 3.35949 0.135232 3.15803Z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
